libbeat/publisher/queue/memqueue/config.go

```diff
@@ -22,32 +22,48 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/elastic/beats/v7/libbeat/common/cfgtype"
 	c "github.com/elastic/elastic-agent-libs/config"
 )
 
 type config struct {
-	Events int `config:"events" validate:"min=32"`
-	// This field is named MaxGetRequest because its logical effect is to give
+	Events *int              `config:"events" validate:"min=32"`
+	Bytes  *cfgtype.ByteSize `config:"bytes"`
+
+	// This field is named MaxGetEvents because its logical effect is to give
 	// a maximum on the number of events a Get request can return, but the
 	// user-exposed name is "flush.min_events" for backwards compatibility,
 	// since it used to control buffer size in the internal buffer chain.
-	MaxGetRequest int           `config:"flush.min_events" validate:"min=0"`
-	FlushTimeout  time.Duration `config:"flush.timeout"`
+	// Ignored if a byte limit is set in the queue or the get request.
+	MaxGetEvents int           `config:"flush.min_events" validate:"min=0"`
+	FlushTimeout time.Duration `config:"flush.timeout"`
 }
 
-var defaultConfig = config{
-	Events:        3200,
-	MaxGetRequest: 1600,
-	FlushTimeout:  10 * time.Second,
-}
+const minQueueBytes = 32768
+const minQueueEvents = 32
+const defaultMaxQueueEvents = 3200
 
 func (c *config) Validate() error {
-	if c.MaxGetRequest > c.Events {
-		return errors.New("flush.min_events must be less events")
+	if c.Bytes != nil && *c.Bytes < minQueueBytes {
```

Comment:
The validation and configuration could be unit tested, it is now more complex than before.